projects/rocr-runtime/rocrtst/suites/functional/counted_queues.cc

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright © Advanced Micro Devices, Inc., or its affiliates.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "suites/functional/counted_queues.h"
+#include "hsa/hsa_ext_amd.h"
+#include "hsa/hsa.h"
+#include "common/base_rocr_utils.h"
+#include "gtest/gtest.h"
+
+
+CountedQueuesTest::CountedQueuesTest() : TestBase() {
+  set_num_iteration(10);  // Number of iterations to execute of the main test;
+                          // This is a default value which can be overridden
+                          // on the command line.
+
+  set_title("RocR Counted Queues Test");
+  set_description(
+      "This test validates the behavior of Shared Counted Queues managed by the "
+      "Counted Queue Manager in a scenario where both HIP and OpenMP utilize CP "
+      "Queues, causing oversubscription and a subsequent performance degradation.");
```

Comment:
In case we have other libraries that decide to use this API, I would omit mentioning HIP and OpenMP. So I would reword as:

"This test validates the behavior of Shared Counted Queues managed by the "
"Counted Queue Manager in a scenario where different libraries use CP "
"Queues, this avoids oversubscription and a subsequent performance degradation.");